### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by xilopaint**

### CODE REVIEW

**Comment by fniephaus** [comment]
Thanks for the PR. It looks like this replaces `alfred-workflow` with `alpynist` and I'm not familiar with it. If someone from the AlfredApp community could provide an update on how workflows handle Python 3 support these days, that'd be great. Also, it'd be great if someone could give this a go and help reviewing. I still have a Python 2 installation on my MacBook.

File: src/workflow/workflow.py [hunk 1]
```diff
@@ -1151,27 +1372,28 @@ def version(self):
...
+                version = self._update_settings.get("version")
 
             # `version` file
             if not version:
-                filepath = self.workflowfile('version')
+                filepath = self.workflowfile("version")
 
                 if os.path.exists(filepath):
-                    with open(filepath, 'rb') as fileobj:
+                    with open(filepath, "rb") as fileobj:
```
Comment (cdw) [inline src/workflow/workflow.py]: Regex `match_version` in `update.py` will fail when given a binary version string like `"b'v8.0'"`. Reading as a text file resolves this.

```suggestion
                    with open(filepath, "r") as fileobj:
```

**Comment by cdw** [comment]
I took a look at the PR and it read ok to me.  Successful install and usage with Alfred 5.0.1 and Python 3.9, no py2 installed. The tests failed until the version file is read in text mode as suggested above.

**Comment by xilopaint** [comment]
The issue was already fixed in the latest version of Alpynist. I just pushed it.

**Comment by devnoname120** [comment]
@xilopaint Any updates on merging this PR?

**Comment by xilopaint** [comment]
> @xilopaint Any updates on merging this PR?

You should ask @fniephaus, I'm only the PR author.

**Comment by fniephaus** [comment]
@devnoname120 Have you tested this PR? I haven't had the time to do so and I'd like to collect some feedback before merging a full rewrite.

**Comment by devnoname120** [comment]
@fniephaus Yes I can confirm that it works. I tested [96169ad](https://github.com/fniephaus/alfred-pocket/pull/61/commits/96169adfde1d7658e35b81d99f809d5a58d46ab8) on Alfred 5.1.

The only awkward part is that the workflow asked me twice to authorize with an error in the middle
In Alfred:
1. I typed `Pocket` in Alfred's bar and authorized it in the web page that opened.
2. I typed again `Pocket` in Alfred's bar but this time it displayed an error. I don't have a screenshot or a log but in the bar it displayed `Error` without any further details.
3. I typed again `Pocket` in Alfred's bar and authorized it again in the web page that opened.
4. I typed again `Pocket` and this time it worked flawlessly, I had access to all my lists and I couldn't find any issues.
5. After the workflow was properly authorized (step 3.) I didn't have any further issues. It now works every time when I open the Alfred bar and type `Pocket`.

File: README.md [hunk 2]
```diff
@@ -16,9 +16,9 @@ This workflow lets you manage your Pocket list with Alfred.
...
 - Supports notifications
 - Uses OAuth 2.0 to authorize the workflow
-- Saves your access_token securely in OS X's keychain
+- Saves your access_token securely in macOS keychain
 
 
 ## Credits
 
-This workflow uses [pocket](https://github.com/tapanpandita/pocket) and [alfred-workflow](https://github.com/deanishe/alfred-workflow).
+This workflow uses [pocket](https://github.com/tapanpandita/pocket) and [Alpynist](https://github.com/xilopaint/alpynist).
```
Comment (devnoname120) [inline README.md:24]: The [link](https://github.com/xilopaint/alpynist) returns a 404 here. Is the repository private?

Comment (xilopaint) [reply README.md:24]: The repository is not published yet, it's only local. The codebase, including the tests, is finished, but I still need to work on README and the documentation. Currently, you can find the latest version of Alpynist in my [Alfred PDF Tools](https://github.com/xilopaint/alfred-pdf-tools) workflow. For now, you can credit me using the link of my profile.

**Comment by devnoname120** [comment]
@fniephaus I haven't thoroughly reviewed the code but imho it's ready for merging.

As it works with recent Alfred versions it's already a big leap forward. Making the code perfect would be a cherry on the cake that can be done in a later iteration and imho shouldn't get in the way of merging this PR. 😊

**Comment by xilopaint** [comment]
@fniephaus Could you make a decision on this PR?

File: README.md (hunk 2 above)
Comment (devnoname120) [reply README.md:24]: Was the repository published since then? The link is still dead.

Comment (xilopaint) [reply README.md:24]: I no longer have plans to publish the repository as I switched from Alfred to Raycast. Alpynist is pretty solid and stable, though, and I'll continue to fix any reported library issues in the Alfred PDF Tools repository.

**Comment by xilopaint** [comment]
Looks like the project is abandoned and there's no interest in supporting Python 3. I'm closing the PR.

**Comment by fniephaus** [comment]
> Looks like the project is abandoned and there's no interest in supporting Python 3

Apologies for the slow response. I'm unfortunately not using Alfred at the moment, but I always welcome contributions. The problem with this PR is that it completely changes the underlying framework. If that framework is somewhat established in the Alfred community, I would be ok with this. At the time this was opened, I didn't see many users of Alpynist. If I accept a PR like this and publish an update, I'd like to at least know what I'm publishing. I hope you understand what I mean.

**Pull request closed without merging**